spec.html

```diff
@@ -497,6 +497,84 @@
   </emu-alg>
 </emu-clause>
 
+<emu-clause id="sec-static-semantics-contains-arguments">
+  <h1>Static Semantics: ContainsArguments</h1>
+  <emu-grammar>
+    IdentifierReference : Identifier
+  </emu-grammar>
+  <emu-alg>
+    1. If the StringValue of |Identifier| is `"arguments"`, return *true*.
+    2. Else, return *false*.
+  </emu-alg>
+
+  <emu-grammar>
+    FunctionDeclaration : `function` BindingIdentifier `(` FormalParameters `)` `{` FunctionBody `}`
+
+    FunctionDeclaration : `function` `(` FormalParameters `)` `{` FunctionBody `}`
+
+    FunctionExpression : `function` BindingIdentifier? `(` FormalParameters `)` `{` FunctionBody `}`
+  </emu-grammar>
+  <emu-alg>
+    1. Return *false*.
+  </emu-alg>
+
+  <emu-grammar>
+    MethodDefinition :
+      PropertyName `(` UniqueFormalParameters `)` `{` FunctionBody `}`
+      `get` PropertyName `(` `)` `{` FunctionBody `}`
+      `set` PropertyName `(` PropertySetParameterList `)` `{` FunctionBody `}`
+  </emu-grammar>
+  <emu-alg>
+    1. Return the result of ContainsArguments for |PropertyName|.
+  </emu-alg>
+
+  <emu-grammar>
+    GeneratorMethod : `*` PropertyName `(` UniqueFormalParameters `)` `{` GeneratorBody `}`
+  </emu-grammar>
+  <emu-alg>
+    1. Return the result of ContainsArguments for |PropertyName|.
+  </emu-alg>
+
+  <emu-grammar>
+    GeneratorDeclaration : `function` `*` BindingIdentifier `(` FormalParameters `)` `{` GeneratorBody `}`
+
+    GeneratorDeclaration : `function` `*` `(` FormalParameters `)` `{` GeneratorBody `}`
+
+    GeneratorExpression : `function` `*` BindingIdentifier? `(` FormalParameters `)` `{` GeneratorBody `}`
+  </emu-grammar>
+  <emu-alg>
+    1. Return *false*.
+  </emu-alg>
+
+  <emu-grammar>ClassBody : ClassElementList</emu-grammar>
+  <emu-alg>
+    1. Return *false*.
+  </emu-alg>
+
+  <emu-grammar>
+    AsyncMethod : `async` [no LineTerminator here] PropertyName `(` UniqueFormalParameters `)` `{` AsyncFunctionBody `}`
+  </emu-grammar>
+  <emu-alg>
+    1. Return the result of ContainsArguments for |PropertyName|.
+  </emu-alg>
+
+  <emu-grammar>
+    AsyncFunctionDeclaration : `async` [no LineTerminator here] `function` BindingIdentifier `(` FormalParameters `)` `{` AsyncFunctionBody `}`
+
+    AsyncFunctionDeclaration : `async` [no LineTerminator here] `function` `(` FormalParameters `)` `{` AsyncFunctionBody `}`
+
+    AsyncFunctionExpression : `async` [no LineTerminator here] `function` `(` FormalParameters `)` `{` AsyncFunctionBody `}`
+
+    AsyncFunctionExpression : `async` [no LineTerminator here] `function` BindingIdentifier `(` FormalParameters `)` `{` AsyncFunctionBody `}`
+  </emu-grammar>
+  <emu-alg>
+    1. Return *false*.
+  </emu-alg>
+
+  For all other grammatical productions, recurse on all nonterminals. If any piece returns *true*, then return *true*. Otherwise return *false*.
+  <emu-note editor>TODO: Elaborate the preceding paragraph with spec text inserted in each relevant place</emu-note>
```

Comment:
I don't think it has to be inserted in many more places; we just need to define this with a single paragraph analogous to Contains in https://tc39.github.io/ecma262/#sec-static-semantic-rules